operator/src/main/java/org/keycloak/operator/controllers/KeycloakService.java

```diff
@@ -44,15 +45,21 @@ public KeycloakService(KubernetesClient client, Keycloak keycloakCR) {
     }
 
     private ServiceSpec getServiceSpec() {
-        String name = isTlsConfigured(keycloak) ? Constants.KEYCLOAK_HTTPS_PORT_NAME : Constants.KEYCLOAK_HTTP_PORT_NAME;
-        return new ServiceSpecBuilder()
-              .addNewPort()
-              .withPort(getServicePort(keycloak))
-              .withName(name)
-              .withProtocol(Constants.KEYCLOAK_SERVICE_PROTOCOL)
-              .endPort()
-              .withSelector(getInstanceLabels())
-              .build();
+        var builder = new ServiceSpecBuilder().withSelector(getInstanceLabels());
+
+        boolean isTlsConfigured = isTlsConfigured(keycloak);
+        if (!isTlsConfigured
+                || Optional.ofNullable(keycloak.getSpec().getHttpSpec()).map(HttpSpec::getHttpEnabled).orElse(false)) {
```

Comment:
I don't really like the comprehensive statements in conditions as it causes worse readability. I'd instead create a separate variable with a proper name. In order to support lazy evaluation, we can use a Supplier (or Predicate). 
@shawkins WDYT?

```suggestion
        boolean isTlsConfigured = isTlsConfigured(keycloak);
        BooleanSupplier isHttpEnabled = () -> Optional.ofNullable(keycloak.getSpec().getHttpSpec()).map(HttpSpec::getHttpEnabled).orElse(false);
        if (!isTlsConfigured || isHttpEnabled.get()) {
```

Reply:
I'm not worried about lazy evaluation here.  Added some intermediate variables to make this more readable